src/assets/wise5/components/common/cRater/CRaterResponse.ts

```diff
@@ -1,18 +1,15 @@
+import { Response } from '../feedbackRule/Response';
 import { CRaterIdea } from './CRaterIdea';
 import { CRaterScore } from './CRaterScore';
 
-export class CRaterResponse {
+export class CRaterResponse extends Response {
   ideas: CRaterIdea[] = [];
   score: number;
   scores: CRaterScore[];
-  submitCounter: number;
 
   constructor(jsonObject: any = {}) {
-    for (const key of Object.keys(jsonObject)) {
-      if (jsonObject[key] != null) {
-        this[key] = jsonObject[key];
-      }
-    }
+    super(jsonObject);
+    this.populateFields(jsonObject);
```

Comment:
When we run the unit tests, because we initialize the ideas variable to an empty array ```ideas: CRaterIdea[] = [];``` in the class declaration, the webpack compiler (I think) adds this line below the line super(jsonObject) ```this.ideas = [];```. If we don't call ```this.populateFields(jsonObject)``` again, the ideas fields is always an empty array, and causes the tests to fail. So in effect, this line is there just to make the tests pass, which is not ideal.

This problem goes away if we don't initialize the ideas variable to an empty array, but this may cause side-effects with the actual behavior of the program.

Reply:
Seems like somewhat unexpected behavior but is good to know for the future.